packages/d3/src/layers.ts

```diff
@@ -64,15 +64,15 @@ function getChartOptions(
 	const height = h || +selection.attr('height') || 0
 	const sOpts = getSelectionOptions(selection, options)
 	return {
-		on: sOpts.on,
+		on: sOpts.on ?? {},
 		attr: {
 			// stick these on for application later
 			width,
 			height,
 			...sOpts.attr,
 		},
-		classed: sOpts.classed,
-		style: sOpts.style,
+		classed: sOpts.classed ?? {},
```

Comment:
These are already supplied as defaults - do we need to make them non-optional to avoid redundant assignment?

Reply:
Updated to use a Partial<> API argument and a non-partial interface internally